assets/templates/dining_lists/dining_slot_diners.html

```diff
@@ -6,48 +6,169 @@
+                                                    <label class="btn btn-outline-primary col-3
+                                                 {% if entry_data.has_shopped %} active {% endif %}">
+                                                        <input type="checkbox" autocomplete="off"
+                                                               id="{{ entry_data.id }}:shop" name="{{ entry_data.id }}:has_shopped"
+                                                                {% if entry_data.has_shopped %} checked {% endif %}>
+                                                        Shop
+                                                    </label>
+                                                    <label class="btn btn-outline-primary col-3
+                                                 {% if entry_data.has_cooked %} active {% endif %}">
+                                                        <input type="checkbox" autocomplete="off"
+                                                               id="{{ entry_data.id }}:cook" name="{{ entry_data.id }}:has_cooked"
+                                                                {% if entry_data.has_cooked %} checked {% endif %}>
+                                                        Cook
+                                                    </label>
+                                                    <label class="btn btn-outline-primary col-3
+                                                 {% if entry_data.has_cleaned %} active {% endif %}">
+                                                        <input type="checkbox" autocomplete="off"
+                                                               id="{{ entry_data.id }}:clean" name="{{ entry_data.id }}:has_cleaned"
+                                                                {% if entry_data.has_cleaned %} checked {% endif %}}>
+                                                        Clean
+                                                    </label>
+                                                {% else %}
+                                                    <div class="col-3">
+                                                        {% if entry_data.has_shopped %} Shop {% endif %}
+                                                    </div>
+                                                    <div class="col-3">
+                                                        {% if entry_data.has_cooked %} Cook {% endif %}
+                                                    </div>
+                                                    <div class="col-3">
+                                                        {% if entry_data.has_cleaned %} Clean {% endif %}
+                                                    </div>
+                                                {% endif %}
+                                            {% else %}
+                                                <div class="col-9"></div>
```

Comment:
I want to present shopcook/clean/pay as a complete group. As most claimers will control both. However, with the unique purchaser it can be that either shop/cook/clean or pay can not be controlled which needs to be visually strong communicated. That is why the structure is why it is. Feel free to design a different layout though if you want to :)

Reply:
I know :wink: it's not about the end result but about how the components are structured in the html. I'm not going to change it, it's just that I'll also not put effort in trying to follow what's happening here.